src/impy/models/dpmjetIII.py

```diff
@@ -136,17 +125,26 @@ def _cross_section(self, kin=None, precision=None):
             if precision is not None:
                 self._lib.dtglgp.jstatb = saved
             return CrossSectionData(inelastic=self._lib.dtglxs.xspro[0, 0, 0])
-        else:
-            stot, sela = self._lib.dt_xshn(
-                self._lib.idt_icihad(kin.p1), self._lib.idt_icihad(kin.p2), 0.0, kin.ecm
-            )
-            return CrossSectionData(total=stot, elastic=sela, inelastic=stot - sela)
+
+        assert kin.p1.is_hadron
+        assert kin.p2.is_hadron
```

Comment:
allowed targets are _projectiles, but in reality it's only `Nuclei()`, proton and neutron.

Reply:
asserts are just an executable comment, I put this here to remind myself that we only deal with hadrons at this point in the function. The nuclear targets are handled by the part above.